src/extension.ts

```diff
@@ -184,18 +207,37 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         synchronize: {
             // Synchronize the setting section 'psalm' to the server (TODO: server side support)
             configurationSection: 'psalm',
-            fileEvents: vscode.workspace.createFileSystemWatcher('**/' + psalmConfigPath)
+            fileEvents: [
+                vscode.workspace.createFileSystemWatcher('**/' + psalmConfigPath),
+                // this is for when files get changed outside of vscode
+                vscode.workspace.createFileSystemWatcher('**/*.php'),
+            ]
         }
     };
     
+    let resolveStartingPromise: undefined|(() => void);
+    vscode.window.setStatusBarMessage("$(extensions) Psalm Language Server is starting ... please wait $(loading~spin)", new Promise((resolve, reject) => {
+        resolveStartingPromise = () => { resolve(); }
+    }));
+
     // Create the language client and start the client.
-    const disposable = new LanguageClient(
+    const lc = new LanguageClient(
         'Psalm Language Server',
         serverOptionsCallbackForDirectory(workspacePath),
         clientOptions
-    ).start();
+    );
 
-    // Push the disposable to the context's subscriptions so that the
+    const disposable = lc.start();
+    
+    // Push the disposable to the context"s subscriptions so that the
     // client can be deactivated on extension deactivation
     context.subscriptions.push(disposable);
+    
+    await lc.onReady();
+
+    // close the loading message in the status bar
+    if (resolveStartingPromise) {
+        resolveStartingPromise();
+    }
+    resolveStartingPromise = undefined;

```

Comment:
After the language client connects and is ready, we hide the loading status on the statusbar.